{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "647ebfd7",
   "metadata": {
    "papermill": {
     "duration": 0.01708,
     "end_time": "2023-08-01T18:28:05.922637",
     "exception": false,
     "start_time": "2023-08-01T18:28:05.905557",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "313375b0",
   "metadata": {
    "papermill": {
     "duration": 0.010015,
     "end_time": "2023-08-01T18:28:05.953412",
     "exception": false,
     "start_time": "2023-08-01T18:28:05.943397",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ff106d16",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-10-14T14:33:37.863523Z",
     "start_time": "2022-10-14T14:33:14.145712Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-01T18:28:05.974970Z",
     "iopub.status.busy": "2023-08-01T18:28:05.974420Z",
     "iopub.status.idle": "2023-08-01T18:28:13.440766Z",
     "shell.execute_reply": "2023-08-01T18:28:13.440017Z"
    },
    "papermill": {
     "duration": 7.480664,
     "end_time": "2023-08-01T18:28:13.443987",
     "exception": false,
     "start_time": "2023-08-01T18:28:05.963323",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization, Metadata\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired, trim_single\n",
    "from qiime2.plugins.demux.methods import filter_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "655b0e6d",
   "metadata": {
    "papermill": {
     "duration": 0.01017,
     "end_time": "2023-08-01T18:28:13.471441",
     "exception": false,
     "start_time": "2023-08-01T18:28:13.461271",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2d9c49e0",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.594341Z",
     "start_time": "2022-06-28T18:50:41.591544Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-01T18:28:13.523220Z",
     "iopub.status.busy": "2023-08-01T18:28:13.522470Z",
     "iopub.status.idle": "2023-08-01T18:28:13.530384Z",
     "shell.execute_reply": "2023-08-01T18:28:13.529569Z"
    },
    "papermill": {
     "duration": 0.021495,
     "end_time": "2023-08-01T18:28:13.532248",
     "exception": false,
     "start_time": "2023-08-01T18:28:13.510753",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'parameter-set-01.yaml')\n",
    "experiment_name = 'exp-01'\n",
    "base_dir = os.path.join('/', 'home', 'username', 'pipeline-dir')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'manifest.csv')\n",
    "replace_files = False\n",
    "trim = None\n",
    "metadata_file = None\n",
    "threads = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e9f167d2",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-01T18:28:13.558342Z",
     "iopub.status.busy": "2023-08-01T18:28:13.557973Z",
     "iopub.status.idle": "2023-08-01T18:28:13.565307Z",
     "shell.execute_reply": "2023-08-01T18:28:13.564211Z"
    },
    "papermill": {
     "duration": 0.020495,
     "end_time": "2023-08-01T18:28:13.567030",
     "exception": false,
     "start_time": "2023-08-01T18:28:13.546535",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "experiment_name = \"thayane-PM-paired-end-trim\"\n",
    "base_dir = \"/home/lauro/nupeb/rede-micro/redemicro-thayane\"\n",
    "manifest_file = (\n",
    "    \"/home/lauro/nupeb/rede-micro/redemicro-thayane/data/manifest-paired.csv\"\n",
    ")\n",
    "metadata_file = (\n",
    "    \"/home/lauro/nupeb/rede-micro/redemicro-thayane/data/paired-end-metadata.tsv\"\n",
    ")\n",
    "class_col = \"above_10\"\n",
    "classifier_file = (\n",
    "    \"/home/lauro/nupeb/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza\"\n",
    ")\n",
    "replace_files = False\n",
    "phred = 20\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n",
    "overlap = 12\n",
    "threads = 6\n",
    "trim = {\n",
    "    \"overlap\": 8,\n",
    "    \"forward_primer\": \"CCTACGGGRSGCAGCAG\",\n",
    "    \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\",\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "73357f61",
   "metadata": {
    "papermill": {
     "duration": 0.010897,
     "end_time": "2023-08-01T18:28:13.591643",
     "exception": false,
     "start_time": "2023-08-01T18:28:13.580746",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7e853e02",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.632125Z",
     "start_time": "2022-06-28T18:50:41.619710Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-01T18:28:13.622541Z",
     "iopub.status.busy": "2023-08-01T18:28:13.622101Z",
     "iopub.status.idle": "2023-08-01T18:28:13.633506Z",
     "shell.execute_reply": "2023-08-01T18:28:13.632327Z"
    },
    "papermill": {
     "duration": 0.033403,
     "end_time": "2023-08-01T18:28:13.635784",
     "exception": false,
     "start_time": "2023-08-01T18:28:13.602381",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New qiime-artifacts folder path created: /home/lauro/nupeb/rede-micro/redemicro-thayane/experiments/thayane-PM-paired-end-trim/qiime-artifacts\n",
      "New img folder path created: /home/lauro/nupeb/rede-micro/redemicro-thayane/experiments/thayane-PM-paired-end-trim/imgs\n"
     ]
    }
   ],
   "source": [
    "# Define the output folder path\n",
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "\n",
    "# Create path if it not exists\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exists\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')\n",
    "\n",
    "# Define the output artifact full path\n",
    "demux_file = os.path.join(out_dir, 'demux-paired.qza')\n",
    "demux_view = os.path.join(out_dir, 'demux-paired.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "614dd2f9",
   "metadata": {
    "papermill": {
     "duration": 0.010905,
     "end_time": "2023-08-01T18:28:13.660485",
     "exception": false,
     "start_time": "2023-08-01T18:28:13.649580",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file of `SampleData[PairedEndSequencesWithQuality]`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c1a14b3b",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-01T18:28:13.683744Z",
     "iopub.status.busy": "2023-08-01T18:28:13.683249Z",
     "iopub.status.idle": "2023-08-01T18:28:13.697379Z",
     "shell.execute_reply": "2023-08-01T18:28:13.696766Z"
    },
    "papermill": {
     "duration": 0.02776,
     "end_time": "2023-08-01T18:28:13.699258",
     "exception": false,
     "start_time": "2023-08-01T18:28:13.671498",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "manifest_df = pd.read_csv(manifest_file)\n",
    "n_directions = len(manifest_df['direction'].unique())\n",
    "if n_directions == 1:\n",
    "    d_type = 'SampleData[SequencesWithQuality]'\n",
    "    v_type = 'SingleEndFastqManifestPhred33'\n",
    "elif n_directions == 2:\n",
    "    d_type = 'SampleData[PairedEndSequencesWithQuality]'\n",
    "    v_type = 'PairedEndFastqManifestPhred33'\n",
    "else:\n",
    "    print(f'ERROR: invalid number of directions {n_directions}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "11e50980",
   "metadata": {
    "collapsed": false,
    "execution": {
     "iopub.execute_input": "2023-08-01T18:28:13.728284Z",
     "iopub.status.busy": "2023-08-01T18:28:13.727880Z",
     "iopub.status.idle": "2023-08-01T18:28:13.749422Z",
     "shell.execute_reply": "2023-08-01T18:28:13.748775Z"
    },
    "papermill": {
     "duration": 0.035984,
     "end_time": "2023-08-01T18:28:13.751414",
     "exception": false,
     "start_time": "2023-08-01T18:28:13.715430",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the metadata file is not defined, use the default metadata file\n",
    "metadata_qa = Metadata.load(metadata_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "9f82334c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:44:58.640338Z",
     "start_time": "2022-06-28T18:50:41.633657Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-01T18:28:13.781003Z",
     "iopub.status.busy": "2023-08-01T18:28:13.780712Z",
     "iopub.status.idle": "2023-08-01T19:28:37.316554Z",
     "shell.execute_reply": "2023-08-01T19:28:37.315915Z"
    },
    "papermill": {
     "duration": 3623.551369,
     "end_time": "2023-08-01T19:28:37.319615",
     "exception": false,
     "start_time": "2023-08-01T18:28:13.768246",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    \n",
    "    # Import data and create an artifact object\n",
    "    artifact = Artifact.import_data(\n",
    "        d_type, \n",
    "        manifest_file, \n",
    "        view_type=v_type)\n",
    "\n",
    "    artifact = filter_samples(demux=artifact, metadata=metadata_qa).filtered_demux\n",
    "    # Save the artifact object to a new qza file\n",
    "    artifact.save(demux_file)\n",
    "\n",
    "else:\n",
    "    artifact = Artifact.load(demux_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "258cf216",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.602855Z",
     "start_time": "2022-06-28T19:44:58.642520Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-01T19:28:37.363163Z",
     "iopub.status.busy": "2023-08-01T19:28:37.362846Z",
     "iopub.status.idle": "2023-08-01T19:30:05.716890Z",
     "shell.execute_reply": "2023-08-01T19:30:05.716182Z"
    },
    "papermill": {
     "duration": 88.382168,
     "end_time": "2023-08-01T19:30:05.719951",
     "exception": false,
     "start_time": "2023-08-01T19:28:37.337783",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "if not os.path.isfile(demux_view) or replace_files: \n",
    "    # Generate e visualization of the Artifact\n",
    "    demux_summary = demux.visualizers.summarize(artifact)\n",
    "\n",
    "    # Save a new visualization file based on the qza file\n",
    "    Visualization.save(demux_summary.visualization, filepath=demux_view)\n",
    "    \n",
    "    demux_view_obj = demux_summary.visualization\n",
    "else:\n",
    "    demux_view_obj = Visualization.load(demux_view)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "05b0a431",
   "metadata": {
    "papermill": {
     "duration": 0.093831,
     "end_time": "2023-08-01T19:30:05.829474",
     "exception": false,
     "start_time": "2023-08-01T19:30:05.735643",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "062fc561",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.609737Z",
     "start_time": "2022-06-28T19:47:56.604820Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-01T19:30:05.932577Z",
     "iopub.status.busy": "2023-08-01T19:30:05.932291Z",
     "iopub.status.idle": "2023-08-01T19:30:05.935866Z",
     "shell.execute_reply": "2023-08-01T19:30:05.935314Z"
    },
    "papermill": {
     "duration": 0.022581,
     "end_time": "2023-08-01T19:30:05.937261",
     "exception": false,
     "start_time": "2023-08-01T19:30:05.914680",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<visualization: Visualization uuid: 4615203b-c3dd-420f-ad7d-77545c5cd848>\n"
     ]
    }
   ],
   "source": [
    "print(demux_view_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "2186d4cb",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.704841Z",
     "start_time": "2022-06-28T19:47:56.611846Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-01T19:30:05.963247Z",
     "iopub.status.busy": "2023-08-01T19:30:05.962972Z",
     "iopub.status.idle": "2023-08-01T19:30:05.969433Z",
     "shell.execute_reply": "2023-08-01T19:30:05.968892Z"
    },
    "papermill": {
     "duration": 0.019825,
     "end_time": "2023-08-01T19:30:05.970825",
     "exception": false,
     "start_time": "2023-08-01T19:30:05.951000",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-grv7xdpq')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 4615203b-c3dd-420f-ad7d-77545c5cd848>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa2214ed",
   "metadata": {
    "papermill": {
     "duration": 0.011999,
     "end_time": "2023-08-01T19:30:05.995103",
     "exception": false,
     "start_time": "2023-08-01T19:30:05.983104",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a86c8027",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-01T19:30:06.019281Z",
     "iopub.status.busy": "2023-08-01T19:30:06.019064Z",
     "iopub.status.idle": "2023-08-01T19:33:10.196634Z",
     "shell.execute_reply": "2023-08-01T19:33:10.195561Z"
    },
    "papermill": {
     "duration": 184.191971,
     "end_time": "2023-08-01T19:33:10.198720",
     "exception": false,
     "start_time": "2023-08-01T19:30:06.006749",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M05_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M05_1_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M05_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M05_1_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M05_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M05_1_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M05_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M05_1_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.36 s (57 µs/read; 1.04 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              6,307\n",
      "  Read 1 with adapter:                   6,057 (96.0%)\n",
      "  Read 2 with adapter:                   5,115 (81.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,379 (21.9%)\n",
      "Pairs written (passing filters):         4,928 (78.1%)\n",
      "\n",
      "Total basepairs processed:     3,216,570 bp\n",
      "  Read 1:     1,923,635 bp\n",
      "  Read 2:     1,292,935 bp\n",
      "Total written (filtered):      2,297,458 bp (71.4%)\n",
      "  Read 1:     1,402,262 bp\n",
      "  Read 2:       895,196 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 6057 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.1\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t1\t0.0\t0\t1\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t916\t0.0\t0\t916\n",
      "19\t1016\t0.0\t0\t1016\n",
      "20\t1108\t0.0\t0\t1108\n",
      "21\t1053\t0.0\t0\t1053\n",
      "22\t1099\t0.0\t0\t1099\n",
      "23\t857\t0.0\t0\t857\n",
      "24\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 5115 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t1\t0.0\t0\t1\n",
      "21\t976\t0.0\t0\t976\n",
      "22\t920\t0.0\t0\t920\n",
      "23\t832\t0.0\t0\t832\n",
      "24\t878\t0.0\t0\t878\n",
      "25\t760\t0.0\t0\t760\n",
      "26\t746\t0.0\t0\t746\n",
      "48\t2\t0.0\t0\t2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M07_34_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M07_35_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M07_34_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M07_35_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M07_34_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M07_35_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M07_34_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M07_35_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.13 s (141 µs/read; 0.43 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:                936\n",
      "  Read 1 with adapter:                     888 (94.9%)\n",
      "  Read 2 with adapter:                     781 (83.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              196 (20.9%)\n",
      "Pairs written (passing filters):           740 (79.1%)\n",
      "\n",
      "Total basepairs processed:       477,360 bp\n",
      "  Read 1:       285,480 bp\n",
      "  Read 2:       191,880 bp\n",
      "Total written (filtered):        344,920 bp (72.3%)\n",
      "  Read 1:       210,509 bp\n",
      "  Read 2:       134,411 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 888 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t127\t0.0\t0\t127\n",
      "19\t172\t0.0\t0\t172\n",
      "20\t155\t0.0\t0\t155\n",
      "21\t150\t0.0\t0\t150\n",
      "22\t158\t0.0\t0\t158\n",
      "23\t124\t0.0\t0\t124\n",
      "46\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 781 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t153\t0.0\t0\t153\n",
      "22\t131\t0.0\t0\t131\n",
      "23\t140\t0.0\t0\t140\n",
      "24\t114\t0.0\t0\t114\n",
      "25\t109\t0.0\t0\t109\n",
      "26\t134\t0.0\t0\t134\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M102_82_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M102_83_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M102_82_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M102_83_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M102_82_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M102_83_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M102_82_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M102_83_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.57 s (16 µs/read; 3.74 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             35,289\n",
      "  Read 1 with adapter:                  34,139 (96.7%)\n",
      "  Read 2 with adapter:                  32,305 (91.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            4,014 (11.4%)\n",
      "Pairs written (passing filters):        31,275 (88.6%)\n",
      "\n",
      "Total basepairs processed:    17,997,390 bp\n",
      "  Read 1:    10,763,145 bp\n",
      "  Read 2:     7,234,245 bp\n",
      "Total written (filtered):     14,579,866 bp (81.0%)\n",
      "  Read 1:     8,898,603 bp\n",
      "  Read 2:     5,681,263 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 34139 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t12\t0.0\t0\t12\n",
      "18\t5146\t0.0\t0\t5146\n",
      "19\t6114\t0.0\t0\t6114\n",
      "20\t6307\t0.0\t0\t6307\n",
      "21\t5716\t0.0\t0\t5716\n",
      "22\t5843\t0.0\t0\t5843\n",
      "23\t4990\t0.0\t0\t4990\n",
      "24\t2\t0.0\t0\t2\n",
      "46\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 32305 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "20\t15\t0.0\t0\t15\n",
      "21\t5991\t0.0\t0\t5991\n",
      "22\t5823\t0.0\t0\t5823\n",
      "23\t5580\t0.0\t0\t5580\n",
      "24\t5463\t0.0\t0\t5463\n",
      "25\t4749\t0.0\t0\t4749\n",
      "26\t4675\t0.0\t0\t4675\n",
      "27\t3\t0.0\t0\t3\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M107_90_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M107_91_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M107_90_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M107_91_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M107_90_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M107_91_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M107_90_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M107_91_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.96 s (12 µs/read; 5.06 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             81,348\n",
      "  Read 1 with adapter:                  78,786 (96.9%)\n",
      "  Read 2 with adapter:                  74,453 (91.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            9,174 (11.3%)\n",
      "Pairs written (passing filters):        72,174 (88.7%)\n",
      "\n",
      "Total basepairs processed:    41,487,480 bp\n",
      "  Read 1:    24,811,140 bp\n",
      "  Read 2:    16,676,340 bp\n",
      "Total written (filtered):     33,646,507 bp (81.1%)\n",
      "  Read 1:    20,536,518 bp\n",
      "  Read 2:    13,109,989 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 78786 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.3\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t20\t0.0\t0\t20\n",
      "18\t12055\t0.0\t0\t12055\n",
      "19\t14119\t0.0\t0\t14119\n",
      "20\t14580\t0.0\t0\t14580\n",
      "21\t13313\t0.0\t0\t13313\n",
      "22\t13401\t0.0\t0\t13401\n",
      "23\t11272\t0.0\t0\t11272\n",
      "24\t7\t0.0\t0\t7\n",
      "25\t2\t0.0\t0\t2\n",
      "27\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 74453 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t35\t0.0\t0\t35\n",
      "21\t13926\t0.0\t0\t13926\n",
      "22\t13307\t0.0\t0\t13307\n",
      "23\t12494\t0.0\t0\t12494\n",
      "24\t12552\t0.0\t0\t12552\n",
      "25\t11277\t0.0\t0\t11277\n",
      "26\t10842\t0.0\t0\t10842\n",
      "27\t7\t0.0\t0\t7\n",
      "38\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M112_96_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M112_97_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M112_96_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M112_97_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M112_96_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M112_97_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M112_96_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M112_97_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.60 s (14 µs/read; 4.14 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             41,405\n",
      "  Read 1 with adapter:                  40,045 (96.7%)\n",
      "  Read 2 with adapter:                  37,893 (91.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            4,723 (11.4%)\n",
      "Pairs written (passing filters):        36,682 (88.6%)\n",
      "\n",
      "Total basepairs processed:    21,116,550 bp\n",
      "  Read 1:    12,628,525 bp\n",
      "  Read 2:     8,488,025 bp\n",
      "Total written (filtered):     17,100,693 bp (81.0%)\n",
      "  Read 1:    10,437,262 bp\n",
      "  Read 2:     6,663,431 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 40045 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t15\t0.0\t0\t15\n",
      "18\t6122\t0.0\t0\t6122\n",
      "19\t7118\t0.0\t0\t7118\n",
      "20\t7358\t0.0\t0\t7358\n",
      "21\t6838\t0.0\t0\t6838\n",
      "22\t6795\t0.0\t0\t6795\n",
      "23\t5792\t0.0\t0\t5792\n",
      "24\t3\t0.0\t0\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 37893 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t13\t0.0\t0\t13\n",
      "21\t7164\t0.0\t0\t7164\n",
      "22\t6767\t0.0\t0\t6767\n",
      "23\t6387\t0.0\t0\t6387\n",
      "24\t6369\t0.0\t0\t6369\n",
      "25\t5653\t0.0\t0\t5653\n",
      "26\t5529\t0.0\t0\t5529\n",
      "27\t6\t0.0\t0\t6\n",
      "29\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M120_104_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M120_105_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M120_104_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M120_105_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M120_104_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M120_105_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M120_104_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M120_105_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.89 s (11 µs/read; 5.51 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            265,830\n",
      "  Read 1 with adapter:                 258,417 (97.2%)\n",
      "  Read 2 with adapter:                 243,378 (91.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           28,950 (10.9%)\n",
      "Pairs written (passing filters):       236,880 (89.1%)\n",
      "\n",
      "Total basepairs processed:   135,573,300 bp\n",
      "  Read 1:    81,078,150 bp\n",
      "  Read 2:    54,495,150 bp\n",
      "Total written (filtered):    110,441,639 bp (81.5%)\n",
      "  Read 1:    67,411,710 bp\n",
      "  Read 2:    43,029,929 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 258417 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t4.1\t0\t5\n",
      "9\t5\t1.0\t0\t5\n",
      "10\t8\t0.3\t0\t8\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t4\t0.0\t0\t4\n",
      "14\t13\t0.0\t0\t13\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t12\t0.0\t0\t12\n",
      "17\t88\t0.0\t0\t88\n",
      "18\t42156\t0.0\t0\t42156\n",
      "19\t46452\t0.0\t0\t46452\n",
      "20\t47307\t0.0\t0\t47307\n",
      "21\t42808\t0.0\t0\t42808\n",
      "22\t42943\t0.0\t0\t42943\n",
      "23\t36585\t0.0\t0\t36585\n",
      "24\t19\t0.0\t0\t19\n",
      "37\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 243378 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t4.1\t0\t1\n",
      "15\t6\t0.0\t0\t6\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t4\t0.0\t0\t4\n",
      "18\t7\t0.0\t0\t7\n",
      "19\t10\t0.0\t0\t10\n",
      "20\t92\t0.0\t0\t92\n",
      "21\t45388\t0.0\t0\t45388\n",
      "22\t44091\t0.0\t0\t44091\n",
      "23\t40805\t0.0\t0\t40805\n",
      "24\t41119\t0.0\t0\t41119\n",
      "25\t36571\t0.0\t0\t36571\n",
      "26\t35248\t0.0\t0\t35248\n",
      "27\t33\t0.0\t0\t33\n",
      "28\t1\t0.0\t0\t1\n",
      "29\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M121_106_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M121_107_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M121_106_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M121_107_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M121_106_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M121_107_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M121_106_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M121_107_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.87 s (11 µs/read; 5.55 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            173,055\n",
      "  Read 1 with adapter:                 168,157 (97.2%)\n",
      "  Read 2 with adapter:                 158,627 (91.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           18,750 (10.8%)\n",
      "Pairs written (passing filters):       154,305 (89.2%)\n",
      "\n",
      "Total basepairs processed:    88,258,050 bp\n",
      "  Read 1:    52,781,775 bp\n",
      "  Read 2:    35,476,275 bp\n",
      "Total written (filtered):     71,939,496 bp (81.5%)\n",
      "  Read 1:    43,911,116 bp\n",
      "  Read 2:    28,028,380 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 168157 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t2.6\t0\t4\n",
      "9\t7\t0.7\t0\t7\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t7\t0.0\t0\t7\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t49\t0.0\t0\t49\n",
      "18\t27234\t0.0\t0\t27234\n",
      "19\t30436\t0.0\t0\t30436\n",
      "20\t30439\t0.0\t0\t30439\n",
      "21\t27745\t0.0\t0\t27745\n",
      "22\t28219\t0.0\t0\t28219\n",
      "23\t23978\t0.0\t0\t23978\n",
      "24\t17\t0.0\t0\t17\n",
      "25\t1\t0.0\t0\t1\n",
      "26\t1\t0.0\t0\t1\n",
      "29\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "35\t1\t0.0\t0\t1\n",
      "36\t1\t0.0\t0\t1\n",
      "44\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 158627 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.2\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t7\t0.0\t0\t7\n",
      "20\t58\t0.0\t0\t58\n",
      "21\t29446\t0.0\t0\t29446\n",
      "22\t28723\t0.0\t0\t28723\n",
      "23\t26316\t0.0\t0\t26316\n",
      "24\t26587\t0.0\t0\t26587\n",
      "25\t24356\t0.0\t0\t24356\n",
      "26\t23106\t0.0\t0\t23106\n",
      "27\t18\t0.0\t0\t18\n",
      "50\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M125_114_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M125_115_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M125_114_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M125_115_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M125_114_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M125_115_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M125_114_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M125_115_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.29 s (10 µs/read; 5.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            221,029\n",
      "  Read 1 with adapter:                 214,652 (97.1%)\n",
      "  Read 2 with adapter:                 202,368 (91.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           24,270 (11.0%)\n",
      "Pairs written (passing filters):       196,759 (89.0%)\n",
      "\n",
      "Total basepairs processed:   112,724,790 bp\n",
      "  Read 1:    67,413,845 bp\n",
      "  Read 2:    45,310,945 bp\n",
      "Total written (filtered):     91,736,264 bp (81.4%)\n",
      "  Read 1:    55,993,983 bp\n",
      "  Read 2:    35,742,281 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 214652 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t4\t0.1\t0\t4\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t9\t0.0\t0\t9\n",
      "17\t56\t0.0\t0\t56\n",
      "18\t34836\t0.0\t0\t34836\n",
      "19\t38914\t0.0\t0\t38914\n",
      "20\t39376\t0.0\t0\t39376\n",
      "21\t35328\t0.0\t0\t35328\n",
      "22\t35700\t0.0\t0\t35700\n",
      "23\t30392\t0.0\t0\t30392\n",
      "24\t16\t0.0\t0\t16\n",
      "25\t1\t0.0\t0\t1\n",
      "33\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 202368 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.8\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t90\t0.0\t0\t90\n",
      "21\t38288\t0.0\t0\t38288\n",
      "22\t36613\t0.0\t0\t36613\n",
      "23\t33625\t0.0\t0\t33625\n",
      "24\t33854\t0.0\t0\t33854\n",
      "25\t30532\t0.0\t0\t30532\n",
      "26\t29321\t0.0\t0\t29321\n",
      "27\t27\t0.0\t0\t27\n",
      "28\t1\t0.0\t0\t1\n",
      "33\t1\t0.0\t0\t1\n",
      "46\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M126_112_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M126_113_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M126_112_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M126_113_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M126_112_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M126_113_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M126_112_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M126_113_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.51 s (11 µs/read; 5.37 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            134,898\n",
      "  Read 1 with adapter:                 131,100 (97.2%)\n",
      "  Read 2 with adapter:                 123,740 (91.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           14,474 (10.7%)\n",
      "Pairs written (passing filters):       120,424 (89.3%)\n",
      "\n",
      "Total basepairs processed:    68,797,980 bp\n",
      "  Read 1:    41,143,890 bp\n",
      "  Read 2:    27,654,090 bp\n",
      "Total written (filtered):     56,143,868 bp (81.6%)\n",
      "  Read 1:    34,269,622 bp\n",
      "  Read 2:    21,874,246 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 131100 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.1\t0\t1\n",
      "11\t3\t0.0\t0\t3\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t45\t0.0\t0\t45\n",
      "18\t21274\t0.0\t0\t21274\n",
      "19\t23591\t0.0\t0\t23591\n",
      "20\t24133\t0.0\t0\t24133\n",
      "21\t21540\t0.0\t0\t21540\n",
      "22\t21620\t0.0\t0\t21620\n",
      "23\t18853\t0.0\t0\t18853\n",
      "24\t21\t0.0\t0\t21\n",
      "28\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 123740 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t7\t0.0\t0\t7\n",
      "20\t43\t0.0\t0\t43\n",
      "21\t22977\t0.0\t0\t22977\n",
      "22\t22378\t0.0\t0\t22378\n",
      "23\t20708\t0.0\t0\t20708\n",
      "24\t20695\t0.0\t0\t20695\n",
      "25\t18788\t0.0\t0\t18788\n",
      "26\t18119\t0.0\t0\t18119\n",
      "27\t18\t0.0\t0\t18\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M133_120_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M133_121_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M133_120_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M133_121_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M133_120_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M133_121_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M133_120_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M133_121_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.54 s (12 µs/read; 5.13 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            131,697\n",
      "  Read 1 with adapter:                 128,018 (97.2%)\n",
      "  Read 2 with adapter:                 120,902 (91.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           14,070 (10.7%)\n",
      "Pairs written (passing filters):       117,627 (89.3%)\n",
      "\n",
      "Total basepairs processed:    67,165,470 bp\n",
      "  Read 1:    40,167,585 bp\n",
      "  Read 2:    26,997,885 bp\n",
      "Total written (filtered):     54,840,992 bp (81.7%)\n",
      "  Read 1:    33,474,559 bp\n",
      "  Read 2:    21,366,433 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 128018 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t2.0\t0\t3\n",
      "10\t2\t0.1\t0\t2\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t9\t0.0\t0\t9\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t36\t0.0\t0\t36\n",
      "18\t20892\t0.0\t0\t20892\n",
      "19\t23217\t0.0\t0\t23217\n",
      "20\t23475\t0.0\t0\t23475\n",
      "21\t21013\t0.0\t0\t21013\n",
      "22\t21297\t0.0\t0\t21297\n",
      "23\t18042\t0.0\t0\t18042\n",
      "24\t16\t0.0\t0\t16\n",
      "28\t2\t0.0\t0\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 120902 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.0\t0\t2\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t38\t0.0\t0\t38\n",
      "21\t22529\t0.0\t0\t22529\n",
      "22\t21826\t0.0\t0\t21826\n",
      "23\t20201\t0.0\t0\t20201\n",
      "24\t20341\t0.0\t0\t20341\n",
      "25\t18341\t0.0\t0\t18341\n",
      "26\t17606\t0.0\t0\t17606\n",
      "27\t10\t0.0\t0\t10\n",
      "48\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M134_122_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M134_123_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M134_122_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M134_123_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M134_122_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M134_123_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M134_122_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M134_123_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.41 s (11 µs/read; 5.31 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            125,104\n",
      "  Read 1 with adapter:                 121,618 (97.2%)\n",
      "  Read 2 with adapter:                 114,529 (91.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           13,643 (10.9%)\n",
      "Pairs written (passing filters):       111,461 (89.1%)\n",
      "\n",
      "Total basepairs processed:    63,803,040 bp\n",
      "  Read 1:    38,156,720 bp\n",
      "  Read 2:    25,646,320 bp\n",
      "Total written (filtered):     51,968,682 bp (81.5%)\n",
      "  Read 1:    31,720,577 bp\n",
      "  Read 2:    20,248,105 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 121618 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.9\t0\t1\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t4\t0.0\t0\t4\n",
      "12\t3\t0.0\t0\t3\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t36\t0.0\t0\t36\n",
      "18\t19986\t0.0\t0\t19986\n",
      "19\t21991\t0.0\t0\t21991\n",
      "20\t22231\t0.0\t0\t22231\n",
      "21\t19944\t0.0\t0\t19944\n",
      "22\t20244\t0.0\t0\t20244\n",
      "23\t17140\t0.0\t0\t17140\n",
      "24\t20\t0.0\t0\t20\n",
      "35\t1\t0.0\t0\t1\n",
      "49\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 114529 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.5\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t39\t0.0\t0\t39\n",
      "21\t21500\t0.0\t0\t21500\n",
      "22\t20970\t0.0\t0\t20970\n",
      "23\t19145\t0.0\t0\t19145\n",
      "24\t19192\t0.0\t0\t19192\n",
      "25\t17137\t0.0\t0\t17137\n",
      "26\t16515\t0.0\t0\t16515\n",
      "27\t15\t0.0\t0\t15\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M141_130_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M141_131_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M141_130_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M141_131_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M141_130_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M141_131_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M141_130_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M141_131_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.40 s (11 µs/read; 5.33 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            124,046\n",
      "  Read 1 with adapter:                 120,595 (97.2%)\n",
      "  Read 2 with adapter:                 112,649 (90.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           14,431 (11.6%)\n",
      "Pairs written (passing filters):       109,615 (88.4%)\n",
      "\n",
      "Total basepairs processed:    63,263,460 bp\n",
      "  Read 1:    37,834,030 bp\n",
      "  Read 2:    25,429,430 bp\n",
      "Total written (filtered):     51,105,654 bp (80.8%)\n",
      "  Read 1:    31,194,473 bp\n",
      "  Read 2:    19,911,181 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 120595 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t3\t0.5\t0\t3\n",
      "10\t2\t0.1\t0\t2\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "14\t9\t0.0\t0\t9\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t8\t0.0\t0\t8\n",
      "17\t36\t0.0\t0\t36\n",
      "18\t19750\t0.0\t0\t19750\n",
      "19\t21663\t0.0\t0\t21663\n",
      "20\t22166\t0.0\t0\t22166\n",
      "21\t19957\t0.0\t0\t19957\n",
      "22\t19855\t0.0\t0\t19855\n",
      "23\t17121\t0.0\t0\t17121\n",
      "24\t17\t0.0\t0\t17\n",
      "25\t1\t0.0\t0\t1\n",
      "49\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 112649 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t42\t0.0\t0\t42\n",
      "21\t21264\t0.0\t0\t21264\n",
      "22\t20187\t0.0\t0\t20187\n",
      "23\t18753\t0.0\t0\t18753\n",
      "24\t18631\t0.0\t0\t18631\n",
      "25\t17217\t0.0\t0\t17217\n",
      "26\t16519\t0.0\t0\t16519\n",
      "27\t20\t0.0\t0\t20\n",
      "62\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M144_134_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M144_135_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M144_134_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M144_135_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M144_134_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M144_135_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M144_134_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M144_135_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.98 s (12 µs/read; 5.11 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             83,748\n",
      "  Read 1 with adapter:                  81,305 (97.1%)\n",
      "  Read 2 with adapter:                  76,468 (91.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            9,433 (11.3%)\n",
      "Pairs written (passing filters):        74,315 (88.7%)\n",
      "\n",
      "Total basepairs processed:    42,711,480 bp\n",
      "  Read 1:    25,543,140 bp\n",
      "  Read 2:    17,168,340 bp\n",
      "Total written (filtered):     34,647,465 bp (81.1%)\n",
      "  Read 1:    21,148,565 bp\n",
      "  Read 2:    13,498,900 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 81305 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t4\t0.1\t0\t4\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t3\t0.0\t0\t3\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t34\t0.0\t0\t34\n",
      "18\t13126\t0.0\t0\t13126\n",
      "19\t14789\t0.0\t0\t14789\n",
      "20\t14919\t0.0\t0\t14919\n",
      "21\t13238\t0.0\t0\t13238\n",
      "22\t13679\t0.0\t0\t13679\n",
      "23\t11493\t0.0\t0\t11493\n",
      "24\t11\t0.0\t0\t11\n",
      "26\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 76468 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t26\t0.0\t0\t26\n",
      "21\t14303\t0.0\t0\t14303\n",
      "22\t13761\t0.0\t0\t13761\n",
      "23\t12641\t0.0\t0\t12641\n",
      "24\t12907\t0.0\t0\t12907\n",
      "25\t11624\t0.0\t0\t11624\n",
      "26\t11189\t0.0\t0\t11189\n",
      "27\t5\t0.0\t0\t5\n",
      "36\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M148_136_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M148_137_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M148_136_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M148_137_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M148_136_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M148_137_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M148_136_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M148_137_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.89 s (13 µs/read; 4.68 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             69,457\n",
      "  Read 1 with adapter:                  67,376 (97.0%)\n",
      "  Read 2 with adapter:                  63,588 (91.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            7,696 (11.1%)\n",
      "Pairs written (passing filters):        61,761 (88.9%)\n",
      "\n",
      "Total basepairs processed:    35,423,070 bp\n",
      "  Read 1:    21,184,385 bp\n",
      "  Read 2:    14,238,685 bp\n",
      "Total written (filtered):     28,794,629 bp (81.3%)\n",
      "  Read 1:    17,575,611 bp\n",
      "  Read 2:    11,219,018 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 67376 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t1.1\t0\t2\n",
      "9\t1\t0.3\t0\t1\n",
      "12\t4\t0.0\t0\t4\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t18\t0.0\t0\t18\n",
      "18\t10806\t0.0\t0\t10806\n",
      "19\t12154\t0.0\t0\t12154\n",
      "20\t12328\t0.0\t0\t12328\n",
      "21\t11299\t0.0\t0\t11299\n",
      "22\t11165\t0.0\t0\t11165\n",
      "23\t9591\t0.0\t0\t9591\n",
      "24\t2\t0.0\t0\t2\n",
      "38\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 63588 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t25\t0.0\t0\t25\n",
      "21\t11966\t0.0\t0\t11966\n",
      "22\t11338\t0.0\t0\t11338\n",
      "23\t10686\t0.0\t0\t10686\n",
      "24\t10803\t0.0\t0\t10803\n",
      "25\t9578\t0.0\t0\t9578\n",
      "26\t9170\t0.0\t0\t9170\n",
      "27\t12\t0.0\t0\t12\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M150_138_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M150_139_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M150_138_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M150_139_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M150_138_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M150_139_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M150_138_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M150_139_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.44 s (11 µs/read; 5.24 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            125,494\n",
      "  Read 1 with adapter:                 121,913 (97.1%)\n",
      "  Read 2 with adapter:                 114,513 (91.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           14,130 (11.3%)\n",
      "Pairs written (passing filters):       111,364 (88.7%)\n",
      "\n",
      "Total basepairs processed:    64,001,940 bp\n",
      "  Read 1:    38,275,670 bp\n",
      "  Read 2:    25,726,270 bp\n",
      "Total written (filtered):     51,920,856 bp (81.1%)\n",
      "  Read 1:    31,692,124 bp\n",
      "  Read 2:    20,228,732 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 121913 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t1.9\t0\t2\n",
      "10\t1\t0.1\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t9\t0.0\t0\t9\n",
      "17\t29\t0.0\t0\t29\n",
      "18\t19784\t0.0\t0\t19784\n",
      "19\t22010\t0.0\t0\t22010\n",
      "20\t22239\t0.0\t0\t22239\n",
      "21\t20205\t0.0\t0\t20205\n",
      "22\t20279\t0.0\t0\t20279\n",
      "23\t17329\t0.0\t0\t17329\n",
      "24\t12\t0.0\t0\t12\n",
      "26\t1\t0.0\t0\t1\n",
      "29\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 114513 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t38\t0.0\t0\t38\n",
      "21\t21285\t0.0\t0\t21285\n",
      "22\t20678\t0.0\t0\t20678\n",
      "23\t19206\t0.0\t0\t19206\n",
      "24\t19254\t0.0\t0\t19254\n",
      "25\t17502\t0.0\t0\t17502\n",
      "26\t16523\t0.0\t0\t16523\n",
      "27\t15\t0.0\t0\t15\n",
      "34\t1\t0.0\t0\t1\n",
      "51\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M151_140_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M151_141_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M151_140_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M151_141_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M151_140_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M151_141_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M151_140_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M151_141_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.94 s (11 µs/read; 5.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            175,547\n",
      "  Read 1 with adapter:                 170,518 (97.1%)\n",
      "  Read 2 with adapter:                 160,113 (91.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           19,868 (11.3%)\n",
      "Pairs written (passing filters):       155,679 (88.7%)\n",
      "\n",
      "Total basepairs processed:    89,528,970 bp\n",
      "  Read 1:    53,541,835 bp\n",
      "  Read 2:    35,987,135 bp\n",
      "Total written (filtered):     72,583,698 bp (81.1%)\n",
      "  Read 1:    44,303,778 bp\n",
      "  Read 2:    28,279,920 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 170518 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.7\t0\t1\n",
      "9\t1\t0.7\t0\t1\n",
      "10\t2\t0.2\t0\t2\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t6\t0.0\t0\t6\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t8\t0.0\t0\t8\n",
      "17\t45\t0.0\t0\t45\n",
      "18\t27718\t0.0\t0\t27718\n",
      "19\t31001\t0.0\t0\t31001\n",
      "20\t31282\t0.0\t0\t31282\n",
      "21\t28036\t0.0\t0\t28036\n",
      "22\t28325\t0.0\t0\t28325\n",
      "23\t24062\t0.0\t0\t24062\n",
      "24\t24\t0.0\t0\t24\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 160113 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t47\t0.0\t0\t47\n",
      "21\t30169\t0.0\t0\t30169\n",
      "22\t28810\t0.0\t0\t28810\n",
      "23\t26859\t0.0\t0\t26859\n",
      "24\t27027\t0.0\t0\t27027\n",
      "25\t24168\t0.0\t0\t24168\n",
      "26\t22998\t0.0\t0\t22998\n",
      "27\t17\t0.0\t0\t17\n",
      "36\t1\t0.0\t0\t1\n",
      "62\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M155_142_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M155_143_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M155_142_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M155_143_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M155_142_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M155_143_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M155_142_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M155_143_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.10 s (10 µs/read; 5.89 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            205,809\n",
      "  Read 1 with adapter:                 199,911 (97.1%)\n",
      "  Read 2 with adapter:                 187,659 (91.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           23,377 (11.4%)\n",
      "Pairs written (passing filters):       182,432 (88.6%)\n",
      "\n",
      "Total basepairs processed:   104,962,590 bp\n",
      "  Read 1:    62,771,745 bp\n",
      "  Read 2:    42,190,845 bp\n",
      "Total written (filtered):     85,055,720 bp (81.0%)\n",
      "  Read 1:    51,917,537 bp\n",
      "  Read 2:    33,138,183 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 199911 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t2\t0.8\t0\t2\n",
      "11\t3\t0.0\t0\t3\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t4\t0.0\t0\t4\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t9\t0.0\t0\t9\n",
      "17\t56\t0.0\t0\t56\n",
      "18\t32726\t0.0\t0\t32726\n",
      "19\t36039\t0.0\t0\t36039\n",
      "20\t36630\t0.0\t0\t36630\n",
      "21\t33071\t0.0\t0\t33071\n",
      "22\t33269\t0.0\t0\t33269\n",
      "23\t28063\t0.0\t0\t28063\n",
      "24\t18\t0.0\t0\t18\n",
      "25\t1\t0.0\t0\t1\n",
      "44\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 187659 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t9\t0.0\t0\t9\n",
      "20\t76\t0.0\t0\t76\n",
      "21\t35021\t0.0\t0\t35021\n",
      "22\t34061\t0.0\t0\t34061\n",
      "23\t31004\t0.0\t0\t31004\n",
      "24\t31672\t0.0\t0\t31672\n",
      "25\t28331\t0.0\t0\t28331\n",
      "26\t27444\t0.0\t0\t27444\n",
      "27\t20\t0.0\t0\t20\n",
      "50\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M156_144_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M156_145_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M156_144_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M156_145_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M156_144_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M156_145_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M156_144_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M156_145_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 4.03 s (10 µs/read; 6.02 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            404,125\n",
      "  Read 1 with adapter:                 392,579 (97.1%)\n",
      "  Read 2 with adapter:                 369,485 (91.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           44,783 (11.1%)\n",
      "Pairs written (passing filters):       359,342 (88.9%)\n",
      "\n",
      "Total basepairs processed:   206,103,750 bp\n",
      "  Read 1:   123,258,125 bp\n",
      "  Read 2:    82,845,625 bp\n",
      "Total written (filtered):    167,535,556 bp (81.3%)\n",
      "  Read 1:   102,260,960 bp\n",
      "  Read 2:    65,274,596 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 392579 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t6.2\t0\t1\n",
      "9\t4\t1.5\t0\t4\n",
      "10\t3\t0.4\t0\t3\n",
      "11\t7\t0.1\t0\t7\n",
      "12\t9\t0.0\t0\t9\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t22\t0.0\t0\t22\n",
      "15\t15\t0.0\t0\t15\n",
      "16\t19\t0.0\t0\t19\n",
      "17\t93\t0.0\t0\t93\n",
      "18\t63372\t0.0\t0\t63372\n",
      "19\t71113\t0.0\t0\t71113\n",
      "20\t71802\t0.0\t0\t71802\n",
      "21\t65233\t0.0\t0\t65233\n",
      "22\t65184\t0.0\t0\t65184\n",
      "23\t55654\t0.0\t0\t55654\n",
      "24\t44\t0.0\t0\t44\n",
      "30\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 369485 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.4\t0\t1\n",
      "11\t1\t0.1\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t15\t0.0\t0\t15\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t8\t0.0\t0\t8\n",
      "19\t6\t0.0\t0\t6\n",
      "20\t114\t0.0\t0\t114\n",
      "21\t69437\t0.0\t0\t69437\n",
      "22\t66510\t0.0\t0\t66510\n",
      "23\t61702\t0.0\t0\t61702\n",
      "24\t62240\t0.0\t0\t62240\n",
      "25\t55721\t0.0\t0\t55721\n",
      "26\t53666\t0.0\t0\t53666\n",
      "27\t53\t0.0\t0\t53\n",
      "29\t1\t0.0\t0\t1\n",
      "32\t1\t0.0\t0\t1\n",
      "47\t1\t0.0\t0\t1\n",
      "48\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M33_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M33_3_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M33_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M33_3_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M33_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M33_3_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M33_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M33_3_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.77 s (13 µs/read; 4.68 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             59,758\n",
      "  Read 1 with adapter:                  57,818 (96.8%)\n",
      "  Read 2 with adapter:                  54,926 (91.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            6,556 (11.0%)\n",
      "Pairs written (passing filters):        53,202 (89.0%)\n",
      "\n",
      "Total basepairs processed:    30,476,580 bp\n",
      "  Read 1:    18,226,190 bp\n",
      "  Read 2:    12,250,390 bp\n",
      "Total written (filtered):     24,802,260 bp (81.4%)\n",
      "  Read 1:    15,137,792 bp\n",
      "  Read 2:     9,664,468 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 57818 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.1\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t14\t0.0\t0\t14\n",
      "18\t8850\t0.0\t0\t8850\n",
      "19\t10210\t0.0\t0\t10210\n",
      "20\t10757\t0.0\t0\t10757\n",
      "21\t9698\t0.0\t0\t9698\n",
      "22\t9797\t0.0\t0\t9797\n",
      "23\t8478\t0.0\t0\t8478\n",
      "24\t5\t0.0\t0\t5\n",
      "33\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 54926 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t16\t0.0\t0\t16\n",
      "21\t10448\t0.0\t0\t10448\n",
      "22\t9857\t0.0\t0\t9857\n",
      "23\t9186\t0.0\t0\t9186\n",
      "24\t9127\t0.0\t0\t9127\n",
      "25\t8316\t0.0\t0\t8316\n",
      "26\t7960\t0.0\t0\t7960\n",
      "27\t5\t0.0\t0\t5\n",
      "28\t1\t0.0\t0\t1\n",
      "31\t2\t0.0\t0\t2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M34_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M34_5_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M34_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M34_5_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M34_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M34_5_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M34_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M34_5_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.70 s (15 µs/read; 4.05 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             47,217\n",
      "  Read 1 with adapter:                  45,581 (96.5%)\n",
      "  Read 2 with adapter:                  43,039 (91.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            5,637 (11.9%)\n",
      "Pairs written (passing filters):        41,580 (88.1%)\n",
      "\n",
      "Total basepairs processed:    24,080,670 bp\n",
      "  Read 1:    14,401,185 bp\n",
      "  Read 2:     9,679,485 bp\n",
      "Total written (filtered):     19,385,224 bp (80.5%)\n",
      "  Read 1:    11,831,636 bp\n",
      "  Read 2:     7,553,588 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 45581 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.2\t0\t1\n",
      "10\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t11\t0.0\t0\t11\n",
      "18\t7184\t0.0\t0\t7184\n",
      "19\t8053\t0.0\t0\t8053\n",
      "20\t8508\t0.0\t0\t8508\n",
      "21\t7521\t0.0\t0\t7521\n",
      "22\t7816\t0.0\t0\t7816\n",
      "23\t6477\t0.0\t0\t6477\n",
      "24\t5\t0.0\t0\t5\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 43039 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t1\t0.0\t0\t1\n",
      "19\t2\t0.0\t0\t2\n",
      "20\t16\t0.0\t0\t16\n",
      "21\t8177\t0.0\t0\t8177\n",
      "22\t7810\t0.0\t0\t7810\n",
      "23\t7237\t0.0\t0\t7237\n",
      "24\t7089\t0.0\t0\t7089\n",
      "25\t6542\t0.0\t0\t6542\n",
      "26\t6152\t0.0\t0\t6152\n",
      "27\t7\t0.0\t0\t7\n",
      "38\t1\t0.0\t0\t1\n",
      "48\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M35_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M35_7_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M35_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M35_7_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M35_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M35_7_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M35_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M35_7_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.01 s (12 µs/read; 4.96 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             83,620\n",
      "  Read 1 with adapter:                  80,722 (96.5%)\n",
      "  Read 2 with adapter:                  76,381 (91.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            9,813 (11.7%)\n",
      "Pairs written (passing filters):        73,807 (88.3%)\n",
      "\n",
      "Total basepairs processed:    42,646,200 bp\n",
      "  Read 1:    25,504,100 bp\n",
      "  Read 2:    17,142,100 bp\n",
      "Total written (filtered):     34,408,130 bp (80.7%)\n",
      "  Read 1:    21,001,219 bp\n",
      "  Read 2:    13,406,911 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 80722 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t3\t0.0\t0\t3\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t23\t0.0\t0\t23\n",
      "18\t12442\t0.0\t0\t12442\n",
      "19\t14366\t0.0\t0\t14366\n",
      "20\t14952\t0.0\t0\t14952\n",
      "21\t13422\t0.0\t0\t13422\n",
      "22\t14006\t0.0\t0\t14006\n",
      "23\t11489\t0.0\t0\t11489\n",
      "24\t6\t0.0\t0\t6\n",
      "34\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 76381 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t20\t0.0\t0\t20\n",
      "21\t14492\t0.0\t0\t14492\n",
      "22\t13547\t0.0\t0\t13547\n",
      "23\t12774\t0.0\t0\t12774\n",
      "24\t12859\t0.0\t0\t12859\n",
      "25\t11418\t0.0\t0\t11418\n",
      "26\t11254\t0.0\t0\t11254\n",
      "27\t8\t0.0\t0\t8\n",
      "28\t1\t0.0\t0\t1\n",
      "30\t2\t0.0\t0\t2\n",
      "33\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M38_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M38_13_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M38_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M38_13_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M38_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M38_13_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M38_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M38_13_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.07 s (12 µs/read; 5.17 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             91,887\n",
      "  Read 1 with adapter:                  88,737 (96.6%)\n",
      "  Read 2 with adapter:                  84,610 (92.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           10,102 (11.0%)\n",
      "Pairs written (passing filters):        81,785 (89.0%)\n",
      "\n",
      "Total basepairs processed:    46,862,370 bp\n",
      "  Read 1:    28,025,535 bp\n",
      "  Read 2:    18,836,835 bp\n",
      "Total written (filtered):     38,126,769 bp (81.4%)\n",
      "  Read 1:    23,270,333 bp\n",
      "  Read 2:    14,856,436 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 88737 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t1.4\t0\t2\n",
      "9\t1\t0.4\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t23\t0.0\t0\t23\n",
      "18\t13511\t0.0\t0\t13511\n",
      "19\t15878\t0.0\t0\t15878\n",
      "20\t16147\t0.0\t0\t16147\n",
      "21\t15011\t0.0\t0\t15011\n",
      "22\t15260\t0.0\t0\t15260\n",
      "23\t12883\t0.0\t0\t12883\n",
      "24\t8\t0.0\t0\t8\n",
      "25\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 84610 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "20\t24\t0.0\t0\t24\n",
      "21\t15873\t0.0\t0\t15873\n",
      "22\t15122\t0.0\t0\t15122\n",
      "23\t14433\t0.0\t0\t14433\n",
      "24\t14084\t0.0\t0\t14084\n",
      "25\t12902\t0.0\t0\t12902\n",
      "26\t12155\t0.0\t0\t12155\n",
      "27\t10\t0.0\t0\t10\n",
      "30\t1\t0.0\t0\t1\n",
      "42\t1\t0.0\t0\t1\n",
      "51\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M39_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M39_15_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M39_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M39_15_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M39_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M39_15_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M39_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M39_15_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 4.13 s (10 µs/read; 6.00 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            413,486\n",
      "  Read 1 with adapter:                 399,769 (96.7%)\n",
      "  Read 2 with adapter:                 378,368 (91.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           47,333 (11.4%)\n",
      "Pairs written (passing filters):       366,153 (88.6%)\n",
      "\n",
      "Total basepairs processed:   210,877,860 bp\n",
      "  Read 1:   126,113,230 bp\n",
      "  Read 2:    84,764,630 bp\n",
      "Total written (filtered):    170,695,250 bp (80.9%)\n",
      "  Read 1:   104,181,918 bp\n",
      "  Read 2:    66,513,332 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 399769 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t4\t1.6\t0\t4\n",
      "10\t5\t0.4\t0\t5\n",
      "11\t3\t0.1\t0\t3\n",
      "12\t7\t0.0\t0\t7\n",
      "13\t5\t0.0\t0\t5\n",
      "14\t21\t0.0\t0\t21\n",
      "15\t16\t0.0\t0\t16\n",
      "16\t22\t0.0\t0\t22\n",
      "17\t93\t0.0\t0\t93\n",
      "18\t61388\t0.0\t0\t61388\n",
      "19\t70621\t0.0\t0\t70621\n",
      "20\t73690\t0.0\t0\t73690\n",
      "21\t66730\t0.0\t0\t66730\n",
      "22\t68998\t0.0\t0\t68998\n",
      "23\t58121\t0.0\t0\t58121\n",
      "24\t35\t0.0\t0\t35\n",
      "25\t1\t0.0\t0\t1\n",
      "27\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "30\t2\t0.0\t0\t2\n",
      "31\t1\t0.0\t0\t1\n",
      "33\t1\t0.0\t0\t1\n",
      "38\t1\t0.0\t0\t1\n",
      "44\t1\t0.0\t0\t1\n",
      "49\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 378368 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.4\t0\t1\n",
      "11\t3\t0.1\t0\t3\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t4\t0.0\t0\t4\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t12\t0.0\t0\t12\n",
      "19\t14\t0.0\t0\t14\n",
      "20\t101\t0.0\t0\t101\n",
      "21\t71516\t0.0\t0\t71516\n",
      "22\t67889\t0.0\t0\t67889\n",
      "23\t63437\t0.0\t0\t63437\n",
      "24\t63388\t0.0\t0\t63388\n",
      "25\t57162\t0.0\t0\t57162\n",
      "26\t54784\t0.0\t0\t54784\n",
      "27\t31\t0.0\t0\t31\n",
      "30\t3\t0.0\t0\t3\n",
      "32\t1\t0.0\t0\t1\n",
      "47\t2\t0.0\t0\t2\n",
      "48\t3\t0.0\t0\t3\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M44_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M44_21_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M44_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M44_21_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M44_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M44_21_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M44_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M44_21_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.72 s (10 µs/read; 5.99 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            371,135\n",
      "  Read 1 with adapter:                 358,569 (96.6%)\n",
      "  Read 2 with adapter:                 340,513 (91.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           41,813 (11.3%)\n",
      "Pairs written (passing filters):       329,322 (88.7%)\n",
      "\n",
      "Total basepairs processed:   189,278,850 bp\n",
      "  Read 1:   113,196,175 bp\n",
      "  Read 2:    76,082,675 bp\n",
      "Total written (filtered):    153,527,611 bp (81.1%)\n",
      "  Read 1:    93,704,549 bp\n",
      "  Read 2:    59,823,062 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 358569 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t5.7\t0\t2\n",
      "9\t2\t1.4\t0\t2\n",
      "10\t9\t0.4\t0\t9\n",
      "11\t5\t0.1\t0\t5\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t27\t0.0\t0\t27\n",
      "15\t12\t0.0\t0\t12\n",
      "16\t18\t0.0\t0\t18\n",
      "17\t107\t0.0\t0\t107\n",
      "18\t55128\t0.0\t0\t55128\n",
      "19\t63838\t0.0\t0\t63838\n",
      "20\t65998\t0.0\t0\t65998\n",
      "21\t60232\t0.0\t0\t60232\n",
      "22\t61611\t0.0\t0\t61611\n",
      "23\t51522\t0.0\t0\t51522\n",
      "24\t47\t0.0\t0\t47\n",
      "25\t1\t0.0\t0\t1\n",
      "26\t1\t0.0\t0\t1\n",
      "27\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "35\t1\t0.0\t0\t1\n",
      "36\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 340513 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t1.4\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t6\t0.0\t0\t6\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t4\t0.0\t0\t4\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t114\t0.0\t0\t114\n",
      "21\t64177\t0.0\t0\t64177\n",
      "22\t61177\t0.0\t0\t61177\n",
      "23\t57422\t0.0\t0\t57422\n",
      "24\t57096\t0.0\t0\t57096\n",
      "25\t51104\t0.0\t0\t51104\n",
      "26\t49340\t0.0\t0\t49340\n",
      "27\t57\t0.0\t0\t57\n",
      "30\t3\t0.0\t0\t3\n",
      "32\t1\t0.0\t0\t1\n",
      "36\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M46_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M46_25_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M46_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M46_25_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M46_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M46_25_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M46_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M46_25_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.97 s (10 µs/read; 6.23 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            204,231\n",
      "  Read 1 with adapter:                 197,843 (96.9%)\n",
      "  Read 2 with adapter:                 186,143 (91.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           23,741 (11.6%)\n",
      "Pairs written (passing filters):       180,490 (88.4%)\n",
      "\n",
      "Total basepairs processed:   104,157,810 bp\n",
      "  Read 1:    62,290,455 bp\n",
      "  Read 2:    41,867,355 bp\n",
      "Total written (filtered):     84,142,438 bp (80.8%)\n",
      "  Read 1:    51,355,131 bp\n",
      "  Read 2:    32,787,307 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 197843 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t3.1\t0\t3\n",
      "9\t4\t0.8\t0\t4\n",
      "10\t2\t0.2\t0\t2\n",
      "11\t3\t0.0\t0\t3\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t11\t0.0\t0\t11\n",
      "17\t48\t0.0\t0\t48\n",
      "18\t30510\t0.0\t0\t30510\n",
      "19\t35181\t0.0\t0\t35181\n",
      "20\t35982\t0.0\t0\t35982\n",
      "21\t33318\t0.0\t0\t33318\n",
      "22\t34098\t0.0\t0\t34098\n",
      "23\t28653\t0.0\t0\t28653\n",
      "24\t19\t0.0\t0\t19\n",
      "31\t1\t0.0\t0\t1\n",
      "33\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 186143 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t3\t0.0\t0\t3\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t6\t0.0\t0\t6\n",
      "20\t67\t0.0\t0\t67\n",
      "21\t35097\t0.0\t0\t35097\n",
      "22\t33354\t0.0\t0\t33354\n",
      "23\t31615\t0.0\t0\t31615\n",
      "24\t31063\t0.0\t0\t31063\n",
      "25\t28240\t0.0\t0\t28240\n",
      "26\t26660\t0.0\t0\t26660\n",
      "27\t24\t0.0\t0\t24\n",
      "30\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "60\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M47_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M47_27_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M47_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M47_27_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M47_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M47_27_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M47_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M47_27_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.10 s (10 µs/read; 5.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            106,627\n",
      "  Read 1 with adapter:                 102,932 (96.5%)\n",
      "  Read 2 with adapter:                  97,552 (91.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,406 (11.6%)\n",
      "Pairs written (passing filters):        94,221 (88.4%)\n",
      "\n",
      "Total basepairs processed:    54,379,770 bp\n",
      "  Read 1:    32,521,235 bp\n",
      "  Read 2:    21,858,535 bp\n",
      "Total written (filtered):     43,924,862 bp (80.8%)\n",
      "  Read 1:    26,809,997 bp\n",
      "  Read 2:    17,114,865 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 102932 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.4\t0\t1\n",
      "10\t2\t0.1\t0\t2\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t27\t0.0\t0\t27\n",
      "18\t15930\t0.0\t0\t15930\n",
      "19\t18254\t0.0\t0\t18254\n",
      "20\t18898\t0.0\t0\t18898\n",
      "21\t17465\t0.0\t0\t17465\n",
      "22\t17595\t0.0\t0\t17595\n",
      "23\t14732\t0.0\t0\t14732\n",
      "24\t12\t0.0\t0\t12\n",
      "29\t2\t0.0\t0\t2\n",
      "46\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 97552 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.6\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "15\t4\t0.0\t0\t4\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t24\t0.0\t0\t24\n",
      "21\t18269\t0.0\t0\t18269\n",
      "22\t17394\t0.0\t0\t17394\n",
      "23\t16603\t0.0\t0\t16603\n",
      "24\t16219\t0.0\t0\t16219\n",
      "25\t14757\t0.0\t0\t14757\n",
      "26\t14265\t0.0\t0\t14265\n",
      "27\t10\t0.0\t0\t10\n",
      "30\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M52_30_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M52_31_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M52_30_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M52_31_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M52_30_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M52_31_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M52_30_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M52_31_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.45 s (15 µs/read; 3.89 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             29,257\n",
      "  Read 1 with adapter:                  28,314 (96.8%)\n",
      "  Read 2 with adapter:                  26,779 (91.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            3,330 (11.4%)\n",
      "Pairs written (passing filters):        25,927 (88.6%)\n",
      "\n",
      "Total basepairs processed:    14,921,070 bp\n",
      "  Read 1:     8,923,385 bp\n",
      "  Read 2:     5,997,685 bp\n",
      "Total written (filtered):     12,086,414 bp (81.0%)\n",
      "  Read 1:     7,377,063 bp\n",
      "  Read 2:     4,709,351 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 28314 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t3\t0.0\t0\t3\n",
      "17\t6\t0.0\t0\t6\n",
      "18\t4262\t0.0\t0\t4262\n",
      "19\t5090\t0.0\t0\t5090\n",
      "20\t5332\t0.0\t0\t5332\n",
      "21\t4660\t0.0\t0\t4660\n",
      "22\t4854\t0.0\t0\t4854\n",
      "23\t4103\t0.0\t0\t4103\n",
      "24\t3\t0.0\t0\t3\n",
      "39\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 26779 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t7\t0.0\t0\t7\n",
      "21\t4977\t0.0\t0\t4977\n",
      "22\t4728\t0.0\t0\t4728\n",
      "23\t4516\t0.0\t0\t4516\n",
      "24\t4566\t0.0\t0\t4566\n",
      "25\t4072\t0.0\t0\t4072\n",
      "26\t3908\t0.0\t0\t3908\n",
      "27\t4\t0.0\t0\t4\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M55_32_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M55_33_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M55_32_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M55_33_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M55_32_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M55_33_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M55_32_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M55_33_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 4.31 s (10 µs/read; 6.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            442,687\n",
      "  Read 1 with adapter:                 428,297 (96.7%)\n",
      "  Read 2 with adapter:                 403,332 (91.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           52,134 (11.8%)\n",
      "Pairs written (passing filters):       390,553 (88.2%)\n",
      "\n",
      "Total basepairs processed:   225,770,370 bp\n",
      "  Read 1:   135,019,535 bp\n",
      "  Read 2:    90,750,835 bp\n",
      "Total written (filtered):    182,075,369 bp (80.6%)\n",
      "  Read 1:   111,128,253 bp\n",
      "  Read 2:    70,947,116 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 428297 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t6.8\t0\t1\n",
      "9\t7\t1.7\t0\t7\n",
      "10\t3\t0.4\t0\t3\n",
      "11\t4\t0.1\t0\t4\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t8\t0.0\t0\t8\n",
      "14\t14\t0.0\t0\t14\n",
      "15\t12\t0.0\t0\t12\n",
      "16\t24\t0.0\t0\t24\n",
      "17\t111\t0.0\t0\t111\n",
      "18\t65665\t0.0\t0\t65665\n",
      "19\t76681\t0.0\t0\t76681\n",
      "20\t79733\t0.0\t0\t79733\n",
      "21\t71733\t0.0\t0\t71733\n",
      "22\t72762\t0.0\t0\t72762\n",
      "23\t61485\t0.0\t0\t61485\n",
      "24\t48\t0.0\t0\t48\n",
      "25\t1\t0.0\t0\t1\n",
      "29\t1\t0.0\t0\t1\n",
      "44\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 403332 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t4\t0.0\t0\t4\n",
      "19\t14\t0.0\t0\t14\n",
      "20\t134\t0.0\t0\t134\n",
      "21\t76031\t0.0\t0\t76031\n",
      "22\t72791\t0.0\t0\t72791\n",
      "23\t68293\t0.0\t0\t68293\n",
      "24\t67095\t0.0\t0\t67095\n",
      "25\t60600\t0.0\t0\t60600\n",
      "26\t58284\t0.0\t0\t58284\n",
      "27\t49\t0.0\t0\t49\n",
      "28\t1\t0.0\t0\t1\n",
      "29\t1\t0.0\t0\t1\n",
      "38\t3\t0.0\t0\t3\n",
      "39\t2\t0.0\t0\t2\n",
      "44\t1\t0.0\t0\t1\n",
      "47\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "60\t1\t0.0\t0\t1\n",
      "62\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M64_44_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M64_45_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M64_44_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M64_45_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M64_44_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M64_45_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M64_44_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M64_45_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 4.38 s (10 µs/read; 6.30 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            459,250\n",
      "  Read 1 with adapter:                 444,346 (96.8%)\n",
      "  Read 2 with adapter:                 417,701 (91.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           54,766 (11.9%)\n",
      "Pairs written (passing filters):       404,484 (88.1%)\n",
      "\n",
      "Total basepairs processed:   234,217,500 bp\n",
      "  Read 1:   140,071,250 bp\n",
      "  Read 2:    94,146,250 bp\n",
      "Total written (filtered):    188,567,425 bp (80.5%)\n",
      "  Read 1:   115,090,590 bp\n",
      "  Read 2:    73,476,835 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 444346 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t7.0\t0\t4\n",
      "9\t2\t1.8\t0\t2\n",
      "10\t3\t0.4\t0\t3\n",
      "11\t7\t0.1\t0\t7\n",
      "12\t5\t0.0\t0\t5\n",
      "14\t13\t0.0\t0\t13\n",
      "15\t15\t0.0\t0\t15\n",
      "16\t8\t0.0\t0\t8\n",
      "17\t97\t0.0\t0\t97\n",
      "18\t67836\t0.0\t0\t67836\n",
      "19\t79427\t0.0\t0\t79427\n",
      "20\t82473\t0.0\t0\t82473\n",
      "21\t74510\t0.0\t0\t74510\n",
      "22\t75842\t0.0\t0\t75842\n",
      "23\t64054\t0.0\t0\t64054\n",
      "24\t44\t0.0\t0\t44\n",
      "25\t1\t0.0\t0\t1\n",
      "32\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "41\t1\t0.0\t0\t1\n",
      "43\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 417701 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t9\t0.0\t0\t9\n",
      "19\t7\t0.0\t0\t7\n",
      "20\t126\t0.0\t0\t126\n",
      "21\t78453\t0.0\t0\t78453\n",
      "22\t75658\t0.0\t0\t75658\n",
      "23\t70104\t0.0\t0\t70104\n",
      "24\t70285\t0.0\t0\t70285\n",
      "25\t62818\t0.0\t0\t62818\n",
      "26\t60160\t0.0\t0\t60160\n",
      "27\t60\t0.0\t0\t60\n",
      "37\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "61\t1\t0.0\t0\t1\n",
      "62\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M72_50_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M72_51_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M72_50_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M72_51_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M72_50_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M72_51_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M72_50_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M72_51_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.79 s (12 µs/read; 4.94 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             64,874\n",
      "  Read 1 with adapter:                  62,613 (96.5%)\n",
      "  Read 2 with adapter:                  58,870 (90.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            7,972 (12.3%)\n",
      "Pairs written (passing filters):        56,902 (87.7%)\n",
      "\n",
      "Total basepairs processed:    33,085,740 bp\n",
      "  Read 1:    19,786,570 bp\n",
      "  Read 2:    13,299,170 bp\n",
      "Total written (filtered):     26,527,193 bp (80.2%)\n",
      "  Read 1:    16,190,715 bp\n",
      "  Read 2:    10,336,478 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 62613 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.2\t0\t1\n",
      "10\t2\t0.1\t0\t2\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t1\t0.0\t0\t1\n",
      "17\t16\t0.0\t0\t16\n",
      "18\t9542\t0.0\t0\t9542\n",
      "19\t11254\t0.0\t0\t11254\n",
      "20\t11593\t0.0\t0\t11593\n",
      "21\t10454\t0.0\t0\t10454\n",
      "22\t10655\t0.0\t0\t10655\n",
      "23\t9084\t0.0\t0\t9084\n",
      "24\t4\t0.0\t0\t4\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 58870 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "20\t15\t0.0\t0\t15\n",
      "21\t11032\t0.0\t0\t11032\n",
      "22\t10603\t0.0\t0\t10603\n",
      "23\t9978\t0.0\t0\t9978\n",
      "24\t9798\t0.0\t0\t9798\n",
      "25\t9012\t0.0\t0\t9012\n",
      "26\t8414\t0.0\t0\t8414\n",
      "27\t12\t0.0\t0\t12\n",
      "51\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M73_52_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M73_53_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M73_52_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M73_53_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M73_52_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M73_53_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M73_52_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M73_53_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.79 s (13 µs/read; 4.51 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             59,161\n",
      "  Read 1 with adapter:                  57,183 (96.7%)\n",
      "  Read 2 with adapter:                  53,857 (91.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            7,067 (11.9%)\n",
      "Pairs written (passing filters):        52,094 (88.1%)\n",
      "\n",
      "Total basepairs processed:    30,172,110 bp\n",
      "  Read 1:    18,044,105 bp\n",
      "  Read 2:    12,128,005 bp\n",
      "Total written (filtered):     24,286,175 bp (80.5%)\n",
      "  Read 1:    14,823,350 bp\n",
      "  Read 2:     9,462,825 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 57183 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t21\t0.0\t0\t21\n",
      "18\t8938\t0.0\t0\t8938\n",
      "19\t10256\t0.0\t0\t10256\n",
      "20\t10571\t0.0\t0\t10571\n",
      "21\t9608\t0.0\t0\t9608\n",
      "22\t9616\t0.0\t0\t9616\n",
      "23\t8157\t0.0\t0\t8157\n",
      "24\t9\t0.0\t0\t9\n",
      "27\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 53857 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t16\t0.0\t0\t16\n",
      "21\t10130\t0.0\t0\t10130\n",
      "22\t9560\t0.0\t0\t9560\n",
      "23\t9121\t0.0\t0\t9121\n",
      "24\t9091\t0.0\t0\t9091\n",
      "25\t8128\t0.0\t0\t8128\n",
      "26\t7791\t0.0\t0\t7791\n",
      "27\t4\t0.0\t0\t4\n",
      "31\t1\t0.0\t0\t1\n",
      "32\t1\t0.0\t0\t1\n",
      "51\t1\t0.0\t0\t1\n",
      "60\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M77_58_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M77_59_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M77_58_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M77_59_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M77_58_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M77_59_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M77_58_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M77_59_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.22 s (11 µs/read; 5.47 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            111,695\n",
      "  Read 1 with adapter:                 107,863 (96.6%)\n",
      "  Read 2 with adapter:                 102,539 (91.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,569 (11.3%)\n",
      "Pairs written (passing filters):        99,126 (88.7%)\n",
      "\n",
      "Total basepairs processed:    56,964,450 bp\n",
      "  Read 1:    34,066,975 bp\n",
      "  Read 2:    22,897,475 bp\n",
      "Total written (filtered):     46,212,008 bp (81.1%)\n",
      "  Read 1:    28,205,331 bp\n",
      "  Read 2:    18,006,677 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 107863 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t1.7\t0\t2\n",
      "9\t1\t0.4\t0\t1\n",
      "10\t2\t0.1\t0\t2\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t25\t0.0\t0\t25\n",
      "18\t16642\t0.0\t0\t16642\n",
      "19\t19084\t0.0\t0\t19084\n",
      "20\t20023\t0.0\t0\t20023\n",
      "21\t18012\t0.0\t0\t18012\n",
      "22\t18353\t0.0\t0\t18353\n",
      "23\t15688\t0.0\t0\t15688\n",
      "24\t9\t0.0\t0\t9\n",
      "32\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 102539 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t28\t0.0\t0\t28\n",
      "21\t19411\t0.0\t0\t19411\n",
      "22\t18348\t0.0\t0\t18348\n",
      "23\t17215\t0.0\t0\t17215\n",
      "24\t17248\t0.0\t0\t17248\n",
      "25\t15532\t0.0\t0\t15532\n",
      "26\t14726\t0.0\t0\t14726\n",
      "27\t15\t0.0\t0\t15\n",
      "30\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "47\t2\t0.0\t0\t2\n",
      "48\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M78_60_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M78_61_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M78_60_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M78_61_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M78_60_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M78_61_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M78_60_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M78_61_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.82 s (10 µs/read; 5.99 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            181,818\n",
      "  Read 1 with adapter:                 175,846 (96.7%)\n",
      "  Read 2 with adapter:                 167,226 (92.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           19,938 (11.0%)\n",
      "Pairs written (passing filters):       161,880 (89.0%)\n",
      "\n",
      "Total basepairs processed:    92,727,180 bp\n",
      "  Read 1:    55,454,490 bp\n",
      "  Read 2:    37,272,690 bp\n",
      "Total written (filtered):     75,465,878 bp (81.4%)\n",
      "  Read 1:    46,061,120 bp\n",
      "  Read 2:    29,404,758 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 175846 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.8\t0\t1\n",
      "9\t2\t0.7\t0\t2\n",
      "10\t5\t0.2\t0\t5\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t7\t0.0\t0\t7\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t9\t0.0\t0\t9\n",
      "17\t46\t0.0\t0\t46\n",
      "18\t26698\t0.0\t0\t26698\n",
      "19\t31601\t0.0\t0\t31601\n",
      "20\t32793\t0.0\t0\t32793\n",
      "21\t29212\t0.0\t0\t29212\n",
      "22\t29914\t0.0\t0\t29914\n",
      "23\t25522\t0.0\t0\t25522\n",
      "24\t27\t0.0\t0\t27\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 167226 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.2\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "15\t6\t0.0\t0\t6\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t5\t0.0\t0\t5\n",
      "20\t52\t0.0\t0\t52\n",
      "21\t31531\t0.0\t0\t31531\n",
      "22\t29981\t0.0\t0\t29981\n",
      "23\t27852\t0.0\t0\t27852\n",
      "24\t27790\t0.0\t0\t27790\n",
      "25\t25434\t0.0\t0\t25434\n",
      "26\t24539\t0.0\t0\t24539\n",
      "27\t25\t0.0\t0\t25\n",
      "30\t1\t0.0\t0\t1\n",
      "38\t1\t0.0\t0\t1\n",
      "47\t1\t0.0\t0\t1\n",
      "48\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M79_62_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M79_63_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M79_62_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M79_63_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M79_62_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M79_63_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M79_62_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M79_63_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.41 s (10 µs/read; 6.28 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            251,880\n",
      "  Read 1 with adapter:                 243,664 (96.7%)\n",
      "  Read 2 with adapter:                 230,412 (91.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           28,790 (11.4%)\n",
      "Pairs written (passing filters):       223,090 (88.6%)\n",
      "\n",
      "Total basepairs processed:   128,458,800 bp\n",
      "  Read 1:    76,823,400 bp\n",
      "  Read 2:    51,635,400 bp\n",
      "Total written (filtered):    104,001,955 bp (81.0%)\n",
      "  Read 1:    63,477,240 bp\n",
      "  Read 2:    40,524,715 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 243664 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t3.8\t0\t1\n",
      "9\t1\t1.0\t0\t1\n",
      "10\t6\t0.2\t0\t6\n",
      "11\t3\t0.1\t0\t3\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t10\t0.0\t0\t10\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t8\t0.0\t0\t8\n",
      "17\t66\t0.0\t0\t66\n",
      "18\t37340\t0.0\t0\t37340\n",
      "19\t43539\t0.0\t0\t43539\n",
      "20\t45309\t0.0\t0\t45309\n",
      "21\t40543\t0.0\t0\t40543\n",
      "22\t41186\t0.0\t0\t41186\n",
      "23\t35616\t0.0\t0\t35616\n",
      "24\t26\t0.0\t0\t26\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 230412 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t1.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t79\t0.0\t0\t79\n",
      "21\t43388\t0.0\t0\t43388\n",
      "22\t41492\t0.0\t0\t41492\n",
      "23\t38580\t0.0\t0\t38580\n",
      "24\t38501\t0.0\t0\t38501\n",
      "25\t34967\t0.0\t0\t34967\n",
      "26\t33354\t0.0\t0\t33354\n",
      "27\t30\t0.0\t0\t30\n",
      "28\t1\t0.0\t0\t1\n",
      "29\t1\t0.0\t0\t1\n",
      "38\t1\t0.0\t0\t1\n",
      "47\t1\t0.0\t0\t1\n",
      "48\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "62\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M88_66_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M88_67_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M88_66_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M88_67_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M88_66_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M88_67_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M88_66_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M88_67_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.42 s (10 µs/read; 6.04 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            243,421\n",
      "  Read 1 with adapter:                 234,893 (96.5%)\n",
      "  Read 2 with adapter:                 221,420 (91.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           29,614 (12.2%)\n",
      "Pairs written (passing filters):       213,807 (87.8%)\n",
      "\n",
      "Total basepairs processed:   124,144,710 bp\n",
      "  Read 1:    74,243,405 bp\n",
      "  Read 2:    49,901,305 bp\n",
      "Total written (filtered):     99,673,922 bp (80.3%)\n",
      "  Read 1:    60,836,006 bp\n",
      "  Read 2:    38,837,916 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 234893 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t3.7\t0\t3\n",
      "9\t1\t0.9\t0\t1\n",
      "10\t4\t0.2\t0\t4\n",
      "11\t7\t0.1\t0\t7\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t10\t0.0\t0\t10\n",
      "15\t12\t0.0\t0\t12\n",
      "16\t12\t0.0\t0\t12\n",
      "17\t56\t0.0\t0\t56\n",
      "18\t35779\t0.0\t0\t35779\n",
      "19\t42124\t0.0\t0\t42124\n",
      "20\t43617\t0.0\t0\t43617\n",
      "21\t39147\t0.0\t0\t39147\n",
      "22\t40034\t0.0\t0\t40034\n",
      "23\t34058\t0.0\t0\t34058\n",
      "24\t22\t0.0\t0\t22\n",
      "55\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 221420 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t3.7\t0\t1\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t5\t0.0\t0\t5\n",
      "18\t4\t0.0\t0\t4\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t72\t0.0\t0\t72\n",
      "21\t41542\t0.0\t0\t41542\n",
      "22\t39768\t0.0\t0\t39768\n",
      "23\t37269\t0.0\t0\t37269\n",
      "24\t37094\t0.0\t0\t37094\n",
      "25\t33440\t0.0\t0\t33440\n",
      "26\t32163\t0.0\t0\t32163\n",
      "27\t34\t0.0\t0\t34\n",
      "47\t4\t0.0\t0\t4\n",
      "48\t2\t0.0\t0\t2\n",
      "50\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "53\t1\t0.0\t0\t1\n",
      "57\t1\t0.0\t0\t1\n",
      "58\t1\t0.0\t0\t1\n",
      "60\t1\t0.0\t0\t1\n",
      "64\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M93_70_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M93_71_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M93_70_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M93_71_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M93_70_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M93_71_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M93_70_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M93_71_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 4.04 s (10 µs/read; 6.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            414,147\n",
      "  Read 1 with adapter:                 400,620 (96.7%)\n",
      "  Read 2 with adapter:                 378,571 (91.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           47,620 (11.5%)\n",
      "Pairs written (passing filters):       366,527 (88.5%)\n",
      "\n",
      "Total basepairs processed:   211,214,970 bp\n",
      "  Read 1:   126,314,835 bp\n",
      "  Read 2:    84,900,135 bp\n",
      "Total written (filtered):    170,876,279 bp (80.9%)\n",
      "  Read 1:   104,295,532 bp\n",
      "  Read 2:    66,580,747 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 400620 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t6.3\t0\t4\n",
      "9\t5\t1.6\t0\t5\n",
      "10\t4\t0.4\t0\t4\n",
      "11\t5\t0.1\t0\t5\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t4\t0.0\t0\t4\n",
      "14\t16\t0.0\t0\t16\n",
      "15\t17\t0.0\t0\t17\n",
      "16\t27\t0.0\t0\t27\n",
      "17\t104\t0.0\t0\t104\n",
      "18\t61978\t0.0\t0\t61978\n",
      "19\t72016\t0.0\t0\t72016\n",
      "20\t74352\t0.0\t0\t74352\n",
      "21\t66806\t0.0\t0\t66806\n",
      "22\t67957\t0.0\t0\t67957\n",
      "23\t57277\t0.0\t0\t57277\n",
      "24\t44\t0.0\t0\t44\n",
      "50\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 378571 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t6.3\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t6\t0.0\t0\t6\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t6\t0.0\t0\t6\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t7\t0.0\t0\t7\n",
      "20\t124\t0.0\t0\t124\n",
      "21\t71295\t0.0\t0\t71295\n",
      "22\t67840\t0.0\t0\t67840\n",
      "23\t63986\t0.0\t0\t63986\n",
      "24\t63309\t0.0\t0\t63309\n",
      "25\t57143\t0.0\t0\t57143\n",
      "26\t54797\t0.0\t0\t54797\n",
      "27\t41\t0.0\t0\t41\n",
      "28\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "32\t1\t0.0\t0\t1\n",
      "38\t1\t0.0\t0\t1\n",
      "46\t1\t0.0\t0\t1\n",
      "48\t1\t0.0\t0\t1\n",
      "49\t1\t0.0\t0\t1\n",
      "64\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M95_72_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M95_73_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M95_72_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M95_73_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M95_72_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M95_73_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M95_72_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M95_73_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.06 s (10 µs/read; 5.93 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            204,057\n",
      "  Read 1 with adapter:                 197,100 (96.6%)\n",
      "  Read 2 with adapter:                 186,765 (91.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           23,508 (11.5%)\n",
      "Pairs written (passing filters):       180,549 (88.5%)\n",
      "\n",
      "Total basepairs processed:   104,069,070 bp\n",
      "  Read 1:    62,237,385 bp\n",
      "  Read 2:    41,831,685 bp\n",
      "Total written (filtered):     84,169,976 bp (80.9%)\n",
      "  Read 1:    51,372,099 bp\n",
      "  Read 2:    32,797,877 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 197100 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t3.1\t0\t3\n",
      "9\t1\t0.8\t0\t1\n",
      "10\t1\t0.2\t0\t1\n",
      "11\t4\t0.0\t0\t4\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t10\t0.0\t0\t10\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t47\t0.0\t0\t47\n",
      "18\t29818\t0.0\t0\t29818\n",
      "19\t34984\t0.0\t0\t34984\n",
      "20\t36896\t0.0\t0\t36896\n",
      "21\t33052\t0.0\t0\t33052\n",
      "22\t33581\t0.0\t0\t33581\n",
      "23\t28674\t0.0\t0\t28674\n",
      "24\t16\t0.0\t0\t16\n",
      "26\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 186765 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.2\t0\t1\n",
      "11\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t52\t0.0\t0\t52\n",
      "21\t35147\t0.0\t0\t35147\n",
      "22\t33796\t0.0\t0\t33796\n",
      "23\t31665\t0.0\t0\t31665\n",
      "24\t30880\t0.0\t0\t30880\n",
      "25\t28167\t0.0\t0\t28167\n",
      "26\t27006\t0.0\t0\t27006\n",
      "27\t26\t0.0\t0\t26\n",
      "29\t1\t0.0\t0\t1\n",
      "30\t2\t0.0\t0\t2\n",
      "38\t1\t0.0\t0\t1\n",
      "47\t1\t0.0\t0\t1\n",
      "48\t1\t0.0\t0\t1\n",
      "56\t1\t0.0\t0\t1\n",
      "63\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M99_78_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M99_79_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M99_78_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M99_79_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M99_78_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-jw2nnpz9/M99_79_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M99_78_L001_R1_001.fastq.gz /tmp/qiime2-archive-zjnprj4v/93655b56-b0ce-4fe7-beb6-76f378ec4020/data/M99_79_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.13 s (12 µs/read; 5.15 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             96,698\n",
      "  Read 1 with adapter:                  93,525 (96.7%)\n",
      "  Read 2 with adapter:                  88,471 (91.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           11,087 (11.5%)\n",
      "Pairs written (passing filters):        85,611 (88.5%)\n",
      "\n",
      "Total basepairs processed:    49,315,980 bp\n",
      "  Read 1:    29,492,890 bp\n",
      "  Read 2:    19,823,090 bp\n",
      "Total written (filtered):     39,911,283 bp (80.9%)\n",
      "  Read 1:    24,359,930 bp\n",
      "  Read 2:    15,551,353 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 93525 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t1.5\t0\t2\n",
      "10\t2\t0.1\t0\t2\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t29\t0.0\t0\t29\n",
      "18\t14577\t0.0\t0\t14577\n",
      "19\t16564\t0.0\t0\t16564\n",
      "20\t17296\t0.0\t0\t17296\n",
      "21\t15656\t0.0\t0\t15656\n",
      "22\t15815\t0.0\t0\t15815\n",
      "23\t13567\t0.0\t0\t13567\n",
      "24\t5\t0.0\t0\t5\n",
      "27\t1\t0.0\t0\t1\n",
      "29\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 88471 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t26\t0.0\t0\t26\n",
      "21\t16805\t0.0\t0\t16805\n",
      "22\t15842\t0.0\t0\t15842\n",
      "23\t14949\t0.0\t0\t14949\n",
      "24\t14599\t0.0\t0\t14599\n",
      "25\t13249\t0.0\t0\t13249\n",
      "26\t12980\t0.0\t0\t12980\n",
      "27\t6\t0.0\t0\t6\n",
      "46\t1\t0.0\t0\t1\n",
      "61\t1\t0.0\t0\t1\n",
      "62\t1\t0.0\t0\t1\n"
     ]
    }
   ],
   "source": [
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')\n",
    "\n",
    "if trim and (not os.path.isfile(demux_file_trim) or replace_files):\n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = None\n",
    "    \n",
    "    if n_directions == 1:\n",
    "        res = trim_single(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front=forward_primer,\n",
    "            adapter=reverse_reverse_complement,\n",
    "            cores=threads,\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    elif n_directions == 2:\n",
    "        res = trim_paired(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front_f=forward_primer,\n",
    "            front_r=reverse_primer,\n",
    "            # adapter_f=reverse_reverse_complement,\n",
    "            # adapter_r=forward_reverse_complement,\n",
    "            cores=threads,\n",
    "            overlap=trim['overlap'],\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    else:\n",
    "        print(f'ERROR: invalid number of directions {n_directions}')\n",
    "\n",
    "    if res is not None:\n",
    "        res.save(demux_file_trim)\n",
    "        Visualization.save(demux.visualizers.summarize(res).visualization, filepath=demux_view_trim)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "1ab3653a",
   "metadata": {
    "collapsed": false,
    "execution": {
     "iopub.execute_input": "2023-08-01T19:33:10.304824Z",
     "iopub.status.busy": "2023-08-01T19:33:10.304377Z",
     "iopub.status.idle": "2023-08-01T19:34:29.481617Z",
     "shell.execute_reply": "2023-08-01T19:34:29.480977Z"
    },
    "papermill": {
     "duration": 79.287726,
     "end_time": "2023-08-01T19:34:29.515953",
     "exception": false,
     "start_time": "2023-08-01T19:33:10.228227",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-5916273s')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: b1763ae8-71b8-4ca8-87d8-7784bf918167>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "demux.visualizers.summarize(res).visualization"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 3985.98927,
   "end_time": "2023-08-01T19:34:30.671593",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-templates/step-prepare-data.ipynb",
   "output_path": "/home/lauro/nupeb/rede-micro/redemicro-thayane/experiments/thayane-PM-paired-end-trim/nb-executed-steps/step-prepare-data-thayane-PM-paired-end-trim.ipynb",
   "parameters": {
    "base_dir": "/home/lauro/nupeb/rede-micro/redemicro-thayane",
    "class_col": "above_10",
    "classifier_file": "/home/lauro/nupeb/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza",
    "experiment_name": "thayane-PM-paired-end-trim",
    "manifest_file": "/home/lauro/nupeb/rede-micro/redemicro-thayane/data/manifest-paired.csv",
    "metadata_file": "/home/lauro/nupeb/rede-micro/redemicro-thayane/data/paired-end-metadata.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2023-08-01T18:28:04.682323",
   "version": "2.3.4"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}